url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/org/jbox2d/dynamics/World.java

```diff
@@ -0,0 +1,1400 @@
+	 * 
+	 * @return the head of the world body list.
+	 */
+	public Body getBodyList() {
+		return m_bodyList;
+	}
+	
+	/**
+	 * Get the world joint list. With the returned joint, use Joint.getNext to get
+	 * the next joint in the world list. A null joint indicates the end of the list.
+	 * 
+	 * @return the head of the world joint list.
+	 */
+	public Joint getJointList() {
+		return m_jointList;
+	}
+	
+	/**
+	 * Get the world contact list. With the returned contact, use Contact.getNext to get
+	 * the next contact in the world list. A null contact indicates the end of the list.
+	 * 
+	 * @return the head of the world contact list.
+	 * @warning contacts are
+	 */
+	public Contact getContactList() {
+		return m_contactManager.m_contactList;
+	}
+	
+	/**
+	 * Enable/disable warm starting. For testing.
+	 * 
+	 * @param flag
+	 */
+	public void setWarmStarting(boolean flag) {
+		m_warmStarting = flag;
+	}
+	
+	public boolean isWarmStarting(){
+		return m_warmStarting;
+	}
+	
+	/**
+	 * Enable/disable continuous physics. For testing.
+	 * 
+	 * @param flag
+	 */
+	public void setContinuousPhysics(boolean flag) {
+		m_continuousPhysics = flag;
+	}
+	
+	public boolean isContinuousPhysics(){
+		return m_continuousPhysics;
+	}
+	
+	/**
+	 * Get the number of broad-phase proxies.
+	 * 
+	 * @return
+	 */
+	public int getProxyCount() {
+		return m_contactManager.m_broadPhase.getProxyCount();
+	}
+	
+	/**
+	 * Get the number of bodies.
+	 * 
+	 * @return
+	 */
+	public int getBodyCount() {
+		return m_bodyCount;
+	}
+	
+	/**
+	 * Get the number of joints.
+	 * 
+	 * @return
+	 */
+	public int getJointCount() {
+		return m_jointCount;
+	}
+	
+	/**
+	 * Get the number of contacts (each may have 0 or more contact points).
+	 * 
+	 * @return
+	 */
+	public int getContactCount() {
+		return m_contactManager.m_contactCount;
+	}
+	
+	/**
+	 * Change the global gravity vector.
+	 * 
+	 * @param gravity
+	 */
+	public void setGravity(Vec2 gravity) {
+		m_gravity.set(gravity);
+	}
+	
+	/**
+	 * Get the global gravity vector.
+	 * 
+	 * @return
+	 */
+	public Vec2 getGravity() {
+		return m_gravity;
+	}
+	
+	/**
+	 * Is the world locked (in the middle of a time step).
+	 * 
+	 * @return
+	 */
+	public boolean isLocked() {
+		return (m_flags & LOCKED) == LOCKED;
+	}
+	
+	/**
+	 * Set flag to control automatic clearing of forces after each time step.
+	 * 
+	 * @param flag
+	 */
+	public void setAutoClearForces(boolean flag) {
+		if (flag) {
+			m_flags |= CLEAR_FORCES;
+		}
+		else {
+			m_flags &= ~CLEAR_FORCES;
+		}
+	}
+	
+	/**
+	 * Get the flag that controls automatic clearing of forces after each time step.
+	 * 
+	 * @return
+	 */
+	public boolean getAutoClearForces() {
+		return (m_flags & CLEAR_FORCES) == CLEAR_FORCES;
+	}
+	
+	private final Island island = new Island();
+	private Body[] stack = new Body[10]; // TODO djm find a good initial stack number;
+	
+	private void solve(TimeStep step) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1013-1025 into a separate method.